lighthouse-core/lib/stack-collector.js

```diff
@@ -51,7 +51,12 @@ async function detectLibraries() {
 
   for (const [name, lib] of Object.entries(libraryDetectorTests)) {
     try {
-      const result = await lib.test(window);
+      /** @type {NodeJS.Timeout|undefined} */
+      let timeout;
+      const timeoutPromise = new Promise(r => timeout = setTimeout(() => r(false), 1000));
```

Comment:
add a comment?  also can it point to https://github.com/GoogleChrome/lighthouse/pull/11172